lib/models/cosmos_wallet.dart

```diff
@@ -0,0 +1,30 @@
+import 'package:sacco/wallet.dart';
+
+class CosmosWallet extends BaseWalletDetails {
+  Wallet wallet;
+
+  CosmosWallet({
+    required String walletAddress,
+    required String walletAlias,
+    WalletType walletType = WalletType.Cosmos,
+    required this.wallet,
+  }) : super(
+          walletAlias: walletAlias,
+          walletAddress: walletAddress,
+          walletType: walletType,
+        );
+}
+
+class BaseWalletDetails {
+  String walletAddress;
+  String walletAlias;
+  WalletType walletType;
+
+  BaseWalletDetails({
+    required this.walletType,
+    required this.walletAlias,
+    required this.walletAddress,
+  });
+}
+
+enum WalletType { Eth, Cosmos }
```

Comment:
this could be moved into separate file aswell

Reply:
✅